src/controllers/createProfile.ts

```diff
@@ -0,0 +1,40 @@
+import type { Context } from 'hono';
+import { StatusCodes } from 'src/constants/status-codes';
+import { SCHEMA_VERSION } from 'src/db/constants';
+import type { Profile } from 'src/types/data/profile';
+
+type CreateProfileRequestBody = Omit<Profile, 'id' | 'schemaVersion'>;
+
+export const createProfile = async (context: Context<EnvironmentBindings>) => {
+	const body: CreateProfileRequestBody = await context.req.json();
+
+	try {
+		// TODO : email, name length, links http format, happpendAt iso format
+		// validateCreateProfileRequestBody(body);
+
+		const id = crypto.randomUUID();
+		const {
+			email,
+			name,
+			description,
+			links,
+			happenedAt
+		} = body;
+		const insertedAt = new Date().toISOString();
+
+		const result = await context.env.database
+			.prepare(`
+				INSERT INTO profile (id, email, name, description, links, happenedAt, insertedAt, schemaVersion)
+				VALUES (?,?,?,?,?,?,?,?)`)
+			.bind(id, email, name, description, links, happenedAt, insertedAt, SCHEMA_VERSION)
+			.run();
+
+		if (result.error) {
```

Comment:
Small correction based on the [docs](https://developers.cloudflare.com/d1/worker-api/return-object/#d1result).

```suggestion
		if (!result.success) {
```